extensions/skins/conversation_v1/Conversation.js

```diff
@@ -18,41 +18,184 @@
+        tutorCardElt.css('margin-left', 'auto');
+        tutorCardElt.css('margin-right', 'auto');
+        tutorCardElt.css('float', '');
+
+        supplementalCardElt.css('opacity', '');
+
+        jQuery(element).removeClass('animate-to-one-card');
+        done();
+      });
+
+      return function(cancel) {
+        if (cancel) {
+          supplementalCardElt.css('opacity', '0');
+          supplementalCardElt.stop();
+
+          tutorCardElt.css('margin-left', '');
+          tutorCardElt.css('margin-right', '');
+          tutorCardElt.css('float', '');
+          tutorCardElt.stop();
+
+          jQuery(element).removeClass('animate-to-one-card');
+        }
+      };
+    } else {
+      return;
+    }
+  };
+
+  return {
+    addClass: animateCards
+  };
+});
+
+
+oppia.animation('.conversation-skin-animate-card-contents', function() {
+  var animateCardChange = function(element, className, done) {
+    if (className !== 'animate-card-change') {
+      return;
+    }
+
+    var currentHeight = element.height();
+    var expectedNextHeight = $('#futurePreview').height();
+
+    // Fix the current card height, so that it does not change during the
+    // animation, even though its contents might.
+    element.css('height', currentHeight);
+
+    jQuery(element).animate({
+      opacity: 0
+    }, TIME_FADEOUT_MSEC).animate({
+      height: expectedNextHeight
+    }, TIME_HEIGHT_CHANGE_MSEC).animate({
+      opacity: 1
+    }, TIME_FADEIN_MSEC, function() {
+      element.css('height', '');
+      done();
+    });
+
+    return function(cancel) {
+      if (cancel) {
+        element.css('opacity', '1.0');
+        element.css('height', '');
+        element.stop();
+      }
+    };
+  };
+
+  return {
+    addClass: animateCardChange
+  };
+});
+
 // TODO(sll): delete/deprecate 'reset exploration' from the list of
 // events sent to a container page.
-
 oppia.directive('conversationSkin', [function() {
   return {
     restrict: 'E',
     scope: {},
     templateUrl: 'skins/Conversation',
     controller: [
-        '$scope', '$timeout', '$rootScope', '$window', '$modal', 'warningsData',
-        'messengerService', 'oppiaPlayerService', 'urlService', 'focusService',
-        'ratingService',
+        '$scope', '$timeout', '$rootScope', '$window', 'messengerService',
+        'oppiaPlayerService', 'urlService', 'focusService', 'ratingService',
+        'windowDimensionsService',
         function(
-          $scope, $timeout, $rootScope, $window, $modal, warningsData,
-          messengerService, oppiaPlayerService, urlService, focusService,
-          ratingService) {
+          $scope, $timeout, $rootScope, $window, messengerService,
+          oppiaPlayerService, urlService, focusService, ratingService,
+          windowDimensionsService) {
+
+      // The minimum width, in pixels, needed to be able to show two cards
+      // side-by-side.
+      $scope.TWO_CARD_THRESHOLD_PX = 1120;
```

Comment:
You mean as a global? I think we try to avoid those in JS, unless you're thinking about declaring it as an angular.constant and then importing it?